python/pyarrow/_dataset_parquet.pyx

```diff
@@ -639,7 +775,9 @@ cdef class ParquetFileWriteOptions(FileWriteOptions):
             write_batch_size=None,
             dictionary_pagesize_limit=None,
             write_page_index=False,
+            encryption_config=None,
```

Comment:
@tolleybot This is called `encryption_config`, but above in the `_set_properties` the keyword that is being set is `self._properties["encryption_properties"]`, so this was actually being ignored?

Also in the test_dataset_encryption.py you added, it is being passed to `encryption_config` keyword.

Reply:
> The enryption_config was not being propagated properly. I have updated the code and included a method _set_encryption_config, to ensure the encryption config is propagated.

Thanks. Do you have an idea why the test was passing then? (I assume because it was ignored, it was just written without encryption, and then reading also works fine, and it will just ignore the decryption config that was passed). Do we have a way to update the test to ensure that it is _actually_ using encryption?

Reply:
Can you check https://github.com/tolleybot/arrow/pull/4 first? In the last commit there, I added a read check without decryption with the assertion that this should error